src/Sulu/Bundle/WebsiteBundle/Sitemap/Provider/PagesSitemapProvider.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\Sitemap\Provider;
+
+use Sulu\Bundle\WebsiteBundle\Sitemap\Sitemap;
+use Sulu\Bundle\WebsiteBundle\Sitemap\SitemapAlternateLink;
+use Sulu\Bundle\WebsiteBundle\Sitemap\SitemapProviderInterface;
+use Sulu\Bundle\WebsiteBundle\Sitemap\SitemapUrl;
+use Sulu\Component\Content\Repository\ContentRepositoryInterface;
+use Sulu\Component\Content\Repository\Mapping\MappingBuilder;
+
+/**
+ * Provides sitemap for webspaces.
+ */
+class PagesSitemapProvider implements SitemapProviderInterface
+{
+    /**
+     * @var ContentRepositoryInterface
+     */
+    private $contentRepository;
+
+    /**
+     * @param ContentRepositoryInterface $contentRepository
+     */
+    public function __construct(ContentRepositoryInterface $contentRepository)
+    {
+        $this->contentRepository = $contentRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate($page, $portalKey, $locale)
+    {
+        $pages = $this->contentRepository->findAllByPortal(
+            $locale,
+            $portalKey,
+            MappingBuilder::create()
+                ->addProperties(['changed', 'seo-hideInSitemap'])
+                ->setResolveUrl(true)
+                ->setHydrateGhost(false)
+                ->getMapping()
+        );
+
+        $result = [];
+        foreach ($pages as $page) {
```

Comment:
This `$page` variable overrides the `$page` argument passed to this method. Might lead to confusing situations if we add pagination in there in the future.